### PR TITLE
added engineId optional parameter to EngineGroup constructor.

### PR DESCRIPTION
**Pull request by detaybey**
The EngineId has to be unique for each application/agent per network. Thus, EngineGroup has to be configurable in this aspect. This commit proposes an optional parameter to the EngineGroup constructor, and sets the EngineId to the provided byte array if not null.

### CODE REVIEW

**Comment by aib** [comment]
Tests fine for me
(In fact, a test that failed on the master branch passed here :)

**Comment by lextm** [comment]
1. Please remove the comment (TODO).
1. Please move `new OctetString(new byte[] { 128, 0, 31, 136, 128, 233, 99, 0, 0, 214, 31, 244 });` to the constructor as well.

**Comment by detaybey** [comment]
- Comment removed,
- Default EngineID values moved under the constructor.

**Comment by lextm** [comment]
You can see both CI builds failed, as line 49 is wrong.

File: SharpSnmpLib/Pipeline/EngineGroup.cs [hunk 1]
```diff
@@ -42,9 +40,14 @@ public sealed class EngineGroup
...
         /// Initializes a new instance of the <see cref="EngineGroup"/> class.
         /// </summary>
-        public EngineGroup()
+        public EngineGroup(byte[] engineId = null)
         {
             _start = DateTime.UtcNow;
+            if(engineId == null) {
+                _engineId = new OctetString(new byte[] { 128, 0, 31, 136, 128, 233, 99, 0, 0, 214, 31, 244 });
+            } else {
+                _engineId = engineId;
```
Comment (lextm) [inline SharpSnmpLib/Pipeline/EngineGroup.cs]: You need to construct an `OctetString` out of `engineId` here.

**Comment by lextm** [comment]
Close this as samples are moved to https://github.com/lextudio/sharpsnmplib-samples

**Pull request closed without merging**